allennlp/nn/beam_search.py

```diff
@@ -327,6 +336,27 @@ def _search(
                 batch_size, self.beam_size * self.per_node_beam_size
             )
 
+            # Track the true probabilities in addition to perturbed probabilities
+            # for use in samplers that may perturb probabilities
+            if self.sampler is not None:
```

Comment:
With the above change, might make sense to check against null value of `last_true_probabilities` here to make mypy happy:

```suggestion
            if last_true_probabilities is not None:
```